src/components/refactor/dialogs/line-creation/line-creation-dialog.js

```diff
@@ -0,0 +1,286 @@
+/**
+ * Copyright (c) 2023, RTE (http://www.rte-france.com)
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+
+import { EQUIPMENT_TYPE, useSnackMessage } from '@gridsuite/commons-ui';
+import { yupResolver } from '@hookform/resolvers/yup';
+import {
+    CONNECTIVITY_1,
+    CONNECTIVITY_2,
+    CURRENT_LIMITS_1,
+    CURRENT_LIMITS_2,
+    EQUIPMENT_ID,
+    EQUIPMENT_NAME,
+    PERMANENT_LIMIT,
+    SERIES_REACTANCE,
+    SERIES_RESISTANCE,
+    SHUNT_CONDUCTANCE_1,
+    SHUNT_CONDUCTANCE_2,
+    SHUNT_SUSCEPTANCE_1,
+    SHUNT_SUSCEPTANCE_2,
+} from 'components/refactor/utils/field-constants';
+import PropTypes from 'prop-types';
+import React, { useCallback, useEffect } from 'react';
+import { FormProvider, useForm } from 'react-hook-form';
+import { useParams } from 'react-router-dom';
+import { createLine } from '../../../../utils/rest-api';
+import { sanitizeString } from '../../../dialogs/dialogUtils';
+import EquipmentSearchDialog from '../../../dialogs/equipment-search-dialog';
+import { useFormSearchCopy } from '../../../dialogs/form-search-copy-hook';
+import { UNDEFINED_CONNECTION_DIRECTION } from '../../../network/constants';
+import yup from '../../utils/yup-config';
+import ModificationDialog from '../commons/modificationDialog';
+import {
+    getConnectivityEmptyFormData,
+    getConnectivityFormData,
+    getConnectivityFormValidationSchema,
+} from '../connectivity/connectivity-form-utils';
+import LineCreationForm from './line-creation-form';
+
+/**
+ * Dialog to create a load in the network
+ * @param currentNodeUuid The node we are currently working on
+ * @param editData the data to edit
+ * @param onCreateLine callback to customize line creation process
+ * @param displayConnectivity to display connectivity section or not
+ * @param voltageLevelOptionsPromise a promise that will bring available voltage levels
+ * @param dialogProps props that are forwarded to the generic ModificationDialog component
+ */
+
+const emptyFormData = {
+    [EQUIPMENT_ID]: null,
+    [EQUIPMENT_NAME]: '',
+    ...getConnectivityEmptyFormData(CONNECTIVITY_1),
+    ...getConnectivityEmptyFormData(CONNECTIVITY_2),
+    [SERIES_RESISTANCE]: null,
+    [SERIES_REACTANCE]: null,
+    [SHUNT_SUSCEPTANCE_1]: null,
+    [SHUNT_CONDUCTANCE_1]: null,
+    [SHUNT_SUSCEPTANCE_2]: null,
+    [SHUNT_CONDUCTANCE_2]: null,
+    [CURRENT_LIMITS_1]: { [PERMANENT_LIMIT]: null },
+    [CURRENT_LIMITS_2]: { [PERMANENT_LIMIT]: null },
+};
+
+const schema = yup
+    .object()
+    .shape({
+        [EQUIPMENT_ID]: yup.string().required(),
+        [EQUIPMENT_NAME]: yup.string(),
+        ...getConnectivityFormValidationSchema(CONNECTIVITY_1),
+        ...getConnectivityFormValidationSchema(CONNECTIVITY_2),
```

Comment:
when displayConnectivity is false, there is a validation error which prevent the user from validating the form
I believe it's because we still add connectivity validation even if we don't display those field

we currently can't create a line in the "attach line" form

Reply:
fixed